experimental/examples/tree-sample/inventory-app/src/main.ts

```diff
@@ -0,0 +1,49 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { DataObject, DataObjectFactory } from "@fluidframework/aqueduct";
+import { ISharedTree, SharedTreeFactory } from "@fluid-internal/tree";
+import { IFluidHandle } from "@fluidframework/core-interfaces";
+import { schema } from "./domains/model";
+
+const treeKey = "tree";
+
+export class Main extends DataObject {
+	public static get Name() {
```

Comment:
My preference is for the factory to own the name rather than the object it creates.  Since it is used for the runtime to find the factory, and not for anyone to find the object.